apps/graduate/src/pages/admin/all/utils/studentDownload/downloadStudentDetailExcel.ts

```diff
@@ -0,0 +1,69 @@
+import * as XLSX from 'xlsx';
+
+import type {
+  StageData,
+  UserDetail,
+} from '~/pages/admin/all/types/allManagement';
+
+export default function downloadStudentDetailExcel(
+  studentDetails: Array<{
+    userDetail: UserDetail & { type: string };
+    stageData: StageData[];
+  }>,
+  filename = '학생 상세 정보.xlsx',
+) {
+  const excelData = studentDetails.map(({ userDetail, stageData }) => {
+    const application = stageData.find(s => s.stage === '신청서');
+    const middleReport = stageData.find(s => s.stage === '중간보고서');
+    const finalReport = stageData.find(s => s.stage === '최종보고서');
+
+    const capstoneStatus = userDetail.etc || '-';
+
+    return {
+      학번: userDetail.studentId,
+      이름: userDetail.name,
+      지도교수: userDetail.professor,
+      졸업년도: userDetail.period,
+      소속학과: userDetail.department,
+      지연횟수: userDetail.delay,
+      캡스톤이수: capstoneStatus,
+      '신청서 상태': application
+        ? application.isSubmit
+          ? userDetail.type
+          : '미제출'
+        : '없음',
+      '중간보고서 상태': middleReport
+        ? middleReport.isSubmit
+          ? '제출'
+          : '미제출'
+        : '없음',
+      '최종보고서 상태': finalReport
+        ? finalReport.isSubmit
+          ? '제출'
+          : '미제출'
+        : '없음',
```

Comment:
```suggestion
    const getSubmissionStatus = (type: '신청서' | '중간보고서' | '최종보고서') => { 
        const statusData = stageData.find(s => s.stage === type);
        if (!statusData) return '없음';
        if (statusData.isSubmit) return '제출';
        if (!statusData.isSubmit) return '미제출';
        return '';
    }
    
    ...
    
      '신청서 상태': getSubmissionStatus('신청서'),
      '중간보고서 상태': getSubmissionStatus('중간보고서'),
      '최종보고서 상태': getSubmissionStatus('최종보고서'),
```

중복 로직은 이런 식으로 함수화하는 것이 좋아 보이네요~

Reply:
이 부분도 아래 말씀드린 타입 적용하면 더 좋겠죠 ㅎㅎ